File: .agents/managed/java/src/main/java/com/example/corporate/security/CryptoException.java
package com.example.corporate.security;

public final class CryptoException extends RuntimeException {
    private static final long serialVersionUID = 1L;
    private final String code;

    public CryptoException(String code, String message) { super(message); this.code = code; }
    public CryptoException(String code, String message, Throwable cause) { super(message, cause); this.code = code; }
    public String code() { return code; }
}

File: Biblioteca_Universal_Corporativa/web-scripting/php/src/CryptoException.php
<?php
declare(strict_types=1);

namespace Mbuc\Security;

final class CryptoException extends \RuntimeException
{
    public function __construct(public readonly string $errorCode, string $message, ?\Throwable $previous = null)
    {
        parent::__construct($message, 0, $previous);
    }
}
